include/conf/load.hpp

```diff
@@ -37,17 +38,60 @@ namespace color_coded
       }
     }
 
-    inline args_t load(std::string const &file, std::string const &filetype)
+    inline args_t load_compilation_database(std::string const &file, fs::path filename)
     {
-      if(file.empty())
-      { return defaults(filetype); }
+      static const std::string source_extensions[] {".c", ".cpp", ".cc"};
+      static const std::string header_extensions[] {".h", ".hpp", ".hh"};
+      std::string error;
+      auto const database_ptr(::clang::tooling::JSONCompilationDatabase::loadFromFile(file, error));
+      if(!database_ptr)
+      { return {}; }
```

Comment:
The only two errors that can arise are IO errors from reading the file and syntax errors in the json file itself. These two are rather unlikely. The search for the file before even calling this function and it is almost always auto-generated by some tool, so there should be no syntax errors. I am not sure how much work it would be to echo the error back to vim.